airflow-core/src/airflow/models/dag.py

```diff
@@ -1613,6 +1623,22 @@ def create_dagrun(
             session=session,
         )
 
+        if dag_deadline := self.get_dagrun_creation_deadlines():
+            Deadline.add_deadline(
```

Comment:
Given all those method does is session.add I would rather we did that directly here. Airflow devs know what session.add does via having an extra method in the way introduces extra cognitive load of "what does this actually do?"

Reply:
Yeah, I can see that I guess.  It felt more "intuitive" this way and allowed for future expansion if we decided we wanted logging or who knows what later.... but maybe I was just overthinking/overengineering it.

How strongly do you feel about this, is it a blocking change or can we leave it to see if anyone else wants to weigh in?

Reply:
I prefer just doing `session.add` too. We do that everywhere else.

Reply:
I think I got it sorted.